packages/navigate/src/links.js

```diff
@@ -1,51 +1,52 @@
 
-let handleLinkClick = () => {}
-let handleLinkHover = () => {}
+export function whenThisLinkIsClicked(el, callback) {
+    el.addEventListener('click', e => {
+        e.preventDefault()
 
-export function whenALinkIsClicked(callback) {
-    handleLinkClick = callback
-
-    initializeLinksForClicking()
+        callback(el)
+    })
 }
 
-export function whenALinkIsHovered(callback) {
-    handleLinkHover = callback
+export function whenThisLinkIsPressed(el, callback) {
+    el.addEventListener('click', e => e.preventDefault())
 
-    initializeLinksForHovering()
-}
+    el.addEventListener('mousedown', e => {
+        e.preventDefault()
 
-export function extractDestinationFromLink(linkEl) {
-    return new URL(linkEl.getAttribute('href'), document.baseURI)
-}
+        callback((whenReleased) => {
+            let handler = e => {
+                e.preventDefault()
 
-export function hijackNewLinksOnThePage() {
-    initializeLinksForClicking()
-    initializeLinksForHovering()
-}
+                whenReleased()
 
-function initializeLinksForClicking() {
-    getLinks().forEach(el => {
-        el.addEventListener('click', e => {
-            e.preventDefault()
+                el.removeEventListener('mouseup', handler)
+            }
 
-            handleLinkClick(el)
+            el.addEventListener('mouseup', handler)
         })
     })
 }
 
-function initializeLinksForHovering() {
-    getLinks()
-        .filter(i => i.hasAttribute('wire:navigate.prefetch'))
-        .forEach(el => {
-            el.addEventListener('mouseenter', e => {
-                handleLinkHover(el)
-            })
-        })
+export function whenThisLinkIsHoveredFor(el, ms = 60, callback) {
+    el.addEventListener('mouseenter', e => {
+        let timeout = setTimeout(() => {
+
+        }, ms)
+
+        let handler = () => {
+            clear
```

Comment:
@calebporzio was this line intentional? Leading to some "Uncaught ReferenceError: clear is not defined"